### ISSUE TITLE
Wrong processing of \, {, } in C/C++ macros if is used with option `texcomments'

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#793]
(Original issue 1086 created by LDVSOFT on 2015-02-17T22:28:08.978423+00:00)

While processing C/C++ macros for LaTeX (via `minted` 2.0) with option `texcomments`, Pygments process backslash `\` and braces `{}` not as a part of C/C++ language but as a start of TeX command. This fails whole typesetting. For example, I want to pygmentize that:

    #define DefineArray(name, type) \
    class name{ \
        type *date; \
    };

adding `texcomments` option will turn that output:

    \begin{Verbatim}[commandchars=\\\{\}]
    \PYGdefault{c+cp}{\PYGdefaultZsh{}define DefineArray(name, type) \PYGdefaultZbs{}}
    \PYGdefault{c+cp}{class name\PYGdefaultZob{} \PYGdefaultZbs{}}
    \PYGdefault{c+cp}{   type *date; \PYGdefaultZbs{}}
    \PYGdefault{c+cp}{\PYGdefaultZcb{};}
    \end{Verbatim}

to that:

    \begin{Verbatim}[commandchars=\\\{\},codes={\catcode`\$=3\catcode`\^=7\catcode`\_=8}]
    \PYGdefault{c+cp}{\PYGdefaultZsh{}define DefineArray(name, type) \}
    \PYGdefault{c+cp}{class name{ \}
    \PYGdefault{c+cp}{   type *date; \}
    \PYGdefault{c+cp}{};}
    \end{Verbatim}

that is obviously wrong.

This happens, thanks to developer of minted package [gpoore](https://github.com/gpoore/minted/issues/66#issuecomment-74721980), because preprocessor token in Pygments is `Comment.Preproc`, and `texcomments` starts working there. The possible solution could be:

* Separate preprocessor to new token. Seems logical because preprocessor code is not actually a comment.
* Add option to ban some `Comment.Preproc`-alike tokens for `texcomments`-like options.

**Comment by LDVSOFT** [comment]
Hi! Any progress?

**Comment by Anteru** [comment]
No, this bug is 4 years old, but I didn't have a change to look at it. If it's an urgent issue for you, would you consider filing a PR? I'm happy to review it.
